src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,121 @@
+                String eventTask = furtherSplitEvent[0];
+                LocalDateTime eventTime = LocalDateTime.parse(furtherSplitEvent[1], Ui.TIME_FORMATTER);
+                output = new EventCommand(eventTask, eventTime);
+                break;
+            } catch (DateTimeParseException e) {
+                output = new UnexpectedCommand("date time parse exception");
+                break;
+            } catch (ArrayIndexOutOfBoundsException e) {
+                output = new UnexpectedCommand("array out of bounds exception, event");
+                break;
+            }
+        case "deadline":
+            try {
+                String[] furtherSplitDeadline = splitText[1].trim().split("/by ");
+                String deadlineTask = furtherSplitDeadline[0];
+                LocalDateTime deadlineBy = LocalDateTime.parse(furtherSplitDeadline[1], Ui.TIME_FORMATTER);
+                output = new DeadlineCommand(deadlineTask, deadlineBy);
+                break;
+            } catch (DateTimeParseException e) {
+                output = new UnexpectedCommand("date time parse exception");
+                break;
+            } catch (ArrayIndexOutOfBoundsException e) {
+                output = new UnexpectedCommand("array out of bounds exception, deadline");
+                break;
+            }
+        case "list":
+            output = new ListCommand();
+            break;
+        case "done":
+            try {
+                int index = Integer.parseInt(splitText[1]);
+                output = new DoneCommand(index);
+                break;
+            } catch (NumberFormatException e) {
+                output = new UnexpectedCommand("number format exception");
+                break;
+            } catch (ArrayIndexOutOfBoundsException e) {
+                output = new UnexpectedCommand("array out of bounds exception");
+                break;
+            }
+        case "delete":
+            try {
+                int index = Integer.parseInt(splitText[1]);
+                output = new DeleteCommand(index);
+                break;
+            } catch (NumberFormatException e) {
+                output = new UnexpectedCommand("number format exception");
+                break;
+            } catch (ArrayIndexOutOfBoundsException e) {
+                output = new UnexpectedCommand("array out of bounds exception");
+                break;
+            }
+        case "find":
+            try {
+                String keyword = splitText[1];
+                output = new FindCommand(keyword);
+                break;
+            } catch (ArrayIndexOutOfBoundsException e) {
+                output = new UnexpectedCommand("array out of bounds exception, find");
+                break;
+            }
+        case "bye":
+            output = new ByeCommand();
+            break;
+        default:
+            output = new UnexpectedCommand("");
+            break;
+        }
+        return output;
```

Comment:
This method could perhaps be shortened if every command was handled separately by individual methods? Up to you ultimately!